velox/common/caching/AsyncDataCache.h

```diff
@@ -216,6 +217,19 @@ class AsyncDataCacheEntry {
 
   void setExclusiveToShared();
 
+  void setSsdFile(SsdFile* FOLLY_NULLABLE file, uint64_t offset) {
```

Comment:
Can this method use RawFileCacheKey or FileCacheKey ? Does it need to pass a pointer to SsdFile?

Can we avoid having SsdFile call this method and create tight coupling between the two? I'm seeing that this method is used in SsdFile::load and SsdFile::write. Can't the caller of these call setSsdFile? SsdFile::write method current doesn't return anything, but it could return a list of offsets for the CachePins it wrote successfully.

Reply:
Looks like this comment got lost. Repeating.

---
Can this method use RawFileCacheKey or FileCacheKey ? Does it need to pass a pointer to SsdFile?

Can we avoid having SsdFile call this method and create tight coupling between the two? I'm seeing that this method is used in SsdFile::load and SsdFile::write. Can't the caller of these call setSsdFile? SsdFile::write method current doesn't return anything, but it could return a list of offsets for the CachePins it wrote successfully.